sympy/functions/combinatorial/factorials.py

```diff
@@ -231,7 +231,7 @@ def _eval_Mod(self, q):
                     else:
                         fc = self._facmod(n, aq)
 
-                    return Integer(fc % q)
```

Comment:
I would not change these -- this is a direct call to the class that needs to handle to value instead of a call to a class that would need to call another.

Reply:
Hello @smichr, I think this is required because just by changing 0 to `S.Zero`, [this](https://github.com/sympy/sympy/blob/f3c949f043adc907179b5dd12e736ecf9a75a9c6/sympy/functions/combinatorial/tests/test_comb_factorials.py#L225) test fails.